File: src/AutoRest.SdkExplorer/Model/Code/VariableDesc.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

using System.Text.Json.Serialization;

namespace AutoRest.SdkExplorer.Model.Code
{
    public class VariableDesc : PlaceHolderDesc
    {
        public TypeDesc? Type { get; set; }

        private VariableDesc()
            : base()
        {
        }

        [JsonConstructor]
        public VariableDesc(string key, string suggestedName, TypeDesc type)
        {
            Key = key;
            SuggestedName = suggestedName;
            Type = type;
        }
    }
}

File: src/AutoRest.SdkExplorer.TypeScript/src/Model/Code/VariableDesc.ts
import { PlaceHolderDesc } from "./PlaceHolderDesc";
import { TypeDesc } from "./TypeDesc";

export class VariableDesc extends PlaceHolderDesc implements AutoRest.SdkExplorer.Interface.VariableDesc {
    type: TypeDesc;

    constructor(data: AutoRest.SdkExplorer.Interface.VariableDesc) {
        super(data);
        this.type = new TypeDesc(data.type!);
    }
}
